workflows/bigstitcher.nf

```diff
@@ -44,39 +33,52 @@ workflow BIGSTITCHER {
 
     main:
 
-    ch_versions = Channel.empty()
+    def ch_versions = Channel.empty()
 
 
     //
     // Create channel from input file and output dir provided through params.input and params.outdir
     //
 
-    Channel.of(output)
-        .map { o ->
+    def ch_data_inputs = Channel.of(output)
+        .multiMap { o ->
             def meta = [ id: "bigstitcher" ]
-            data_files = []
-            if (xml_file) {
-                data_files << xml_file
+            def data_files = []
+            def module_args = []
+            if (params.xml) {
+                if (is_local_file(params.xml)) {
+                    // only add it as a file if it's a local file
+                    data_files << param_as_file(params.xml)
+                }
+                // xml inputs are always passed using '-x' flag
+                module_args << '-x' << param_as_file(params.xml)
             }
-            if (o) {
-                data_files << o
+            if (is_local_file(o)) {
+                data_files << param_as_file(o)
             }
-            def input_data = [
-                meta,
-                data_files,
-            ]
-            log.debug "Input data: ${module_class} ${input_data} ${module_params}"
-            // return input_data
-            input_data
+            // outputs are always passed using '-o' flag
+            module_args << '-o' << param_as_file(o)
+
+            if (module_params) {
+                module_args << module_params
+            }
+
+            if (params.input_data_files) {
+                get_values_as_collection(params.input_data_files).forEach { f ->
+                    data_files << param_as_file(f)
+                }
+            }
+
+            log.debug "BigStitcher input: ${meta}, ${module.module_class} ${module_args}, data files: ${data_files}"
+            data_inputs: [ meta, data_files ]
+            module_inputs: [ meta, module.module_class, module_args ]
```

Comment:
@krokicki  and @StephanPreibisch we could add to this to extract files that need to be be mounted automatically in the container directly from the xml or from the OME attributes. If we can do that - then users will not have to pass in the extra '--input_data_files' arg